pkg/reconciler/taskrun/taskrun.go

```diff
@@ -294,8 +295,20 @@ func (c *Reconciler) finishReconcileUpdateEmitEvents(ctx context.Context, tr *v1
 
 	afterCondition := tr.Status.GetCondition(apis.ConditionSucceeded)
 
-	// Send k8s events and cloud events (when configured)
-	events.Emit(ctx, beforeCondition, afterCondition, tr)
+	if afterCondition.IsFalse() && !tr.IsCancelled() && tr.IsRetriable() {
+		tr.Status.CompletionTime = &metav1.Time{Time: c.Clock.Now()}
+		newStatus := tr.Status.DeepCopy()
+		newStatus.RetriesStatus = nil
+		tr.Status.RetriesStatus = append(tr.Status.RetriesStatus, *newStatus)
+		tr.Status.StartTime = nil
+		tr.Status.PodName = ""
+		taskRunCondSet := apis.NewBatchConditionSet()
+		taskRunCondSet.Manage(&tr.Status).MarkUnknown(apis.ConditionSucceeded, "CompletedWithRetries", "")
```

Comment:
let's make the reason a constant `TaskRunReason` so it's easier to reuse (and change later)

https://github.com/tektoncd/pipeline/blob/83af16733e00b7572d2156e73b8266c71766a43e/pkg/apis/pipeline/v1beta1/taskrun_types.go#L155-L184